artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java

```diff
@@ -443,6 +443,14 @@ public ActiveMQThreadFactory run() {
 
             logger.debug("Acceptor using native kqueue");
          } else {
+            if (useEpoll) {
+               ActiveMQServerLogger.LOGGER.nettyEpollNotAvailable(name);
+            }
+
+            if (useKQueue) {
+               ActiveMQServerLogger.LOGGER.nettyKQueueNotAvailable(name);
+            }
```

Comment:
I'll change it to raise the warning only if the `useEpoll` parameter is explicitly set

Reply:
Its true by default and is configured true again in the broker.xml by default, so a bunch of folks will have that explicitly set already. If its going to be a warning you will have to do all the brittle platform checks as well, which per other comment there are still cases that still wont be sufficient and a warning will be emitted in cases its expected not to work.

I really dont think it should be logging a warning for this.

Reply:
@gemmellr I would like to add a warning for 2 reasons:

- Using epoll may greatly improve performance (so switching to NIO could cause performance issues)
- If users explicitly set `useEpoll=true` and broker switch to NIO the users should know

I could check the ENV or remove `useEpoll=true` from default config to mitigate noising warnings for new installations, WDYT?

Reply:
If you are going to have warnings I think youll probably have to check the env being used just to cut down on the erroneous logs during cases it can never work, but yes the default xml config should change too.

That said, folks love to copy existing config forward to new installs and so many might still have it explicitly set even though it doenst apply to them in their use, especially since it has never really been a 'will use epoll' setting and is more of a 'can use, in the cases it can work'. So I think thats still going to lead to enquiries from the folks it incorrectly logs a warning for. On the flip side it also seems likely many wont set it explicitly themselves for totally pristine new installs as they either just don't know about it or have never needed to (and still wont) change it given it defaults to true, meaning the warnings wouldnt show for them anyway.

Its also fairly typical for folks to have configs that might be used in multiple different environments which arent necessarily theirs (though having disparate development and testing/prod environments is also actually pretty common), so for them they would have to avoid setting it explicitly to avoid people getting erroneous warnings in the envs its not expected to and cant work (though adding env checks would reduce those). This is probably more typical on the client side though, where I'd guess it is less likely to be set explicitly though (which would mean the warnings wont be seen). up there

I think it would be simpler all round just to just always log what _is_ being used, at least for the broker where its a one-off event vs every-connection for the client, where what you propose might be better (though wont actually be seen unless explicitly toggled).

Reply:
ATM I'm not able to figure out a better way to notify users so I'll drop warnings